File: trunk/java/SR-RBD/src/ddb/db/TableRow.java
package ddb.db;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * Przechowuje wartosci wiersza tabeli. Przechowuje rowniez referencje do tabeli,
 * co pozwala na pobieranie wartosci wg nazwy kolumny.
 * @author mateusz
 */
public class TableRow implements Serializable {
    /**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	/**
     * Utworz wiersz do tabeli
     * @param table tabela, do ktorej nalezy wiersz
     * @param values wartosci w wierszu, w takiej kolejnosci, jak nazwy kolumn
     * w tabeli
     */
    public TableRow(DatabaseTable table, ArrayList<Object> values) {
        this.table = table;
        this.values = values;
        if(table.getColumnCount() != values.size()){
            throw new IndexOutOfBoundsException("table.getColumnCount() != values.size()");
        }
    }

    public ArrayList<Object> getValues() {
        return values;
    }

    public DatabaseTable getTable() {
        return table;
    }

    /**
     * Pobierz wartość z danej kolumny
     * @param columnIndex numer kolumny
     * @return wartosc w tej kolumnie
     */
    public Object getValue(int columnIndex) {
        return values.get(columnIndex);
    }
    
    /** Wartosci w poszczegolnych kolumnach. */
    private ArrayList<Object> values;
    private DatabaseTable table;
}

File: C#/RBD/RBD/DB/TableRow.cs
// +

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace RBD.DB
{
    [Serializable]
    public class TableRow
    {
        /**
         * Utworz wiersz do tabeli
         * @param table tabela, do ktorej nalezy wiersz
         * @param values wartosci w wierszu, w takiej kolejnosci, jak nazwy kolumn
         * w tabeli
         */
        public TableRow(DatabaseTable table, IList<Object> values)
        {
            this.table = table;
            this.values = values;
            if (table.getColumnCount() != values.Count)
            {
                throw new IndexOutOfRangeException("table.getColumnCount() != values.size()");
            }
        }

        public IList<Object> getValues()
        {
            return values;
        }

        public DatabaseTable getTable()
        {
            return table;
        }

        /**
         * Pobierz wartość z danej kolumny
         * @param columnIndex numer kolumny
         * @return wartosc w tej kolumnie
         */
        public Object getValue(int columnIndex)
        {
            return values[columnIndex];
        }

        override
        public String ToString()
        {
            StringBuilder builder = new StringBuilder();

            foreach (Object v in getValues())
            {
                builder.Append(v);
                builder.Append("\t");
            }

            return builder.ToString();

        }

        /** Wartosci w poszczegolnych kolumnach. */
        private IList<Object> values;
        private DatabaseTable table;
    }

}

